lib/index.js

```diff
@@ -10,23 +10,41 @@ const Joi = require('joi');
 const Schema = require('./schema');
 const Package = require('../package.json');
 
-// const SchwiftyMigration = require('schwifty-migration');
-
 const internals = {};
 
 exports.register = function (server, options, next) {
 
+    Joi.assert(options, Schema.plugin, 'Bad plugin options passed to schwifty.');
+
     const rootState = internals.state(server.root);
 
-    internals.state(server).dir = options.dir;
     const knexFile = options.knexFile;
 
-    internals.state(server).knex = Knex(knexFile[process.env.NODE_ENV]);
+    /*
+        Move the models into the knexFile object
+        Since a models path isn't part of it's schema on knexjs.org
+    */
+
+    if (options.models) {
+        knexFile.models = options.models;
+    }
+
+    let knex;
```

Comment:
> Since we're supporting multiple plugins so multiple DB's in that way, I don't have a strong need for it.

Can you explain this?  I don't see how we support multiple DBs if there's only one knex instance for the entire hapi server, but maybe I'm missing something.